packages/shopify-codemod/transforms/console-utils.js

```diff
@@ -0,0 +1,68 @@
+import chalk from 'chalk';
+
+const NEWLINE_REGEX = /\r\n|[\n\v\f\r\x85\u2028\u2029]/;
+const CONTEXT_LINES = 2;
+
+function padLeft(text, length) {
+  return (' '.repeat(length) + text).slice(-length);
+}
+
+function paintRange(lineNum, startLine, startColumn, endLine, endColumn) {
+  let paintStart = Number.MAX_SAFE_INTEGER;
+  let paintEnd = Number.MAX_SAFE_INTEGER;
+
+  if (startLine <= lineNum && lineNum <= endLine) {
+    if (lineNum === startLine) {
+      paintStart = startColumn;
+    } else {
+      paintStart = 0;
+    }
+    if (lineNum === endLine) {
+      paintEnd = endColumn;
+    } else {
+      paintEnd = Number.MAX_SAFE_INTEGER;
+    }
+  }
+  return {paintStart, paintEnd};
+}
+
+function paintLine(line, from, to, color) {
+  return `${line.slice(0, from)}${color(line.slice(from, to))}${line.slice(to)}`;
+}
+
+function formatLine(line, lineNum) {
+  return `${chalk.gray(padLeft(lineNum + 1, 4))}|${line}\n`;
+}
+
+function codeContext(node, source, color) {
```

Comment:
This is better, but still does too much for me.  My brain is occupied with splitting the "figure out which lines to display" from the "display the lines" parts.

So figuring out the lines is probably better in its own function, like:
```js
function getContextLines(loc, source) {
  const lines = source.split(NEWLINE_REGEX);
  const {start, end} = loc;
  const startLineIndex = loc.start.line - 1;
  const endLineIndex = loc.end.line - 1;

  const contextStart = Math.max(0, startLineIndex - CONTEXT_LINES);
  const contextEnd = Math.min(lines.length, endLineIndex + CONTEXT_LINES);

  return lines.slice(contextStart, contextEnd).map((line, i) => {
    const lineIndex = i + contextStart;
    const isHighlighted = (lineIndex >= startLineIndex) && (lineIndex <= endLineIndex);

    return {
      line,
      lineIndex,
      isHighlighted,
      ...getHighlightRange(isHighlighted, start.column, end.column),
    };
  });
}


function getHighlightRange(isHighlighted, startColumn, endColumn) {
  const highlightStart = isHighlighted ? startColumn : Number.MAX_SAFE_INTEGER;
  const highlightEnd = isHighlighted ? endColumn : Number.MAX_SAFE_INTEGER;

  return {highlightStart, highlightEnd};
}
```

Note that the result contains more than just strings.  That allows us to simplify display a little:

```js
function codeContext(node, source, color) {
  const lines = getContextLines(node.get('loc').value, source);

  return lines.map((lineInfo) =>
    formatLine(
      paintLine(lineInfo, color),
      lineInfo.lineIndex + 1,
    ),
  ).join('');
}

function paintLine({line, highlightStart, highlightEnd}, color) {
  const prefix = line.slice(0, highlightStart);
  const highlighted = line.slice(highlightStart, highlightEnd);
  const suffix = line.slice(highlightEnd);

  return `${prefix}${color(highlighted)}${suffix}`;
}
```